common_tools/get_dumps.py

```diff
@@ -0,0 +1,40 @@
+import os
+from pathlib import Path
+import argparse
+
+
+def compile_dumps(mlir_dir: str):
+    dumps_dir = Path(args.dir).parent.joinpath("dumps")
+    i_o_paths = []
+    for root, _, files in os.walk(mlir_dir):
+        for file in files:
+            if not file.endswith(".mlir") and not file.endswith(".mlirbc"):
+                continue
```

Comment:
nit: use `filter` and get the extension from the path object
```suggestion
        for mlir_file in filter(lambda fn: fn.suffix in (".mlir", ".mlirbc"), files):
```